server/datastore/mysql/hosts.go

```diff
@@ -51,17 +50,41 @@ func (d *Datastore) NewHost(ctx context.Context, host *fleet.Host) (*fleet.Host,
 		host.OSVersion,
 		host.Uptime,
 		host.Memory,
-		host.SeenTime,
 		host.TeamID,
 	)
 	if err != nil {
 		return nil, errors.Wrap(err, "new host")
 	}
 	id, _ := result.LastInsertId()
 	host.ID = uint(id)
+
+	_, err = d.writer.ExecContext(ctx, `INSERT INTO host_seen_times (host_id, seen_time) VALUES (?,?)`, host.ID, host.SeenTime)
+	if err != nil {
+		return nil, errors.Wrap(err, "new host seen time")
+	}
+
 	return host, nil
 }
 
+func (d *Datastore) SerialSaveHost(ctx context.Context, host *fleet.Host) error {
```

Comment:
Well, it might not be for this particular test we ran, but unbounded concurrency is problematic anyway, so I would rather leave this as an option.

I will test it out with this disabled and get some numbers for tomorrow.

Reply:
+1 to testing w/o serialized writes.

If we really need this I would try always waiting for the serial-save-goroutine to finish before returning from all the methods where it's used.

Reply:
I think at some point we'll have to seriously consider adding a message queue config option instead of building ad hoc asynchronous solutions. As Tomas mentions, it's definitely true that unbounded concurrency leads to issues and add enough hosts and we'll probably get a bunch of places with such issues. That being said, it's not a quick fix so to get this particular case to work we could start with this PR and deprecate it down the road when it can be better handled with a queue. If however the test without async save host works fine, I'd punt on adding this async support until actually needed - if we end up adding message queue support first, it might never be needed in this form.

Reply:
Spoiler alert: without serialized writes + new seen time doesn't look great :)